src/states/establishing_node.rs

```diff
@@ -39,52 +38,69 @@ use std::{
 
 const POKE_TIMEOUT: Duration = Duration::from_secs(60);
 
+pub struct EstablishingNodeDetails {
+    pub ack_mgr: AckManager,
+    pub cache: Box<Cache>,
+    pub crust_service: Service,
+    pub full_id: FullId,
+    pub gen_pfx_info: GenesisPfxInfo,
+    pub min_section_size: usize,
+    pub msg_backlog: Vec<RoutingMessage>,
+    pub notified_nodes: BTreeSet<PublicId>,
+    pub peer_mgr: PeerManager,
+    pub routing_msg_filter: RoutingMessageFilter,
+    pub timer: Timer,
+}
+
 pub struct EstablishingNode {
-    pub(super) ack_mgr: AckManager,
-    pub(super) cache: Box<Cache>,
-    pub(super) chain: Chain,
-    pub(super) crust_service: Service,
-    pub(super) full_id: FullId,
-    pub(super) gen_pfx_info: GenesisPfxInfo,
+    ack_mgr: AckManager,
```

Comment:
Wouldn't it make sense to just use composition to avoid repeating all these fields?